### PR TITLE
Age prediction using machine learning tools in Galaxy

### PR DESCRIPTION
**Pull request by anuprulez**
This PR adds a training material to showcase that similar accuracies for chronological age prediction can be achieved using machine learning tools in Galaxy as published in a couple of recent scientific studies.
The datasets, RNA-seq and DNA methylation, have been taken from these studies:
- RNA-seq dataset from [Jason G. Fleischer et al. 2018](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1599-6#Sec9)
- DNA methylation dataset from [Jana Naue et al. 2017](https://www.sciencedirect.com/science/article/pii/S1872497317301643?via%3Dihub)

@polkhe thanks a lot for your contribution!

### CODE REVIEW

File: topics/statistics/tutorials/age-prediction_with_ml/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+
+In the previous step, we generated predictions for the test dataset. We have one more dataset (`test_rows_labels`) which contains the true age values of the test set. Using the true and predicted values of age in the test set, we will verify the performance by analysing the plots.
+
+> ### {% icon hands_on %} Hands-on: Create regression plots
+>
+> 1. **Plot actual vs predicted curves and residual plots of tabular data** {% icon tool %} with the following parameters:
+>    - {% icon param-files %} *"Select input data file :"*: `test_rows_labels` tabular file
+>    - {% icon param-files %} *"Select predicted data file :"*: `Tabular` file (output of **Ensemble methods for classification and regression** {% icon tool %})
+>
+>    > ### {% icon comment %} Comment
```
Comment (bebatut) [inline topics/statistics/tutorials/age-prediction_with_ml/tutorial.md]: You could move all the description of the outputs and the figures outside of the hands-on box (which should mostly serve to describe sth the reader has to do)

File: topics/statistics/tutorials/age-prediction_with_ml/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+>    > and predicted  age. For a good model, there should not be any visible pattern with the plotted points.
+>    >
+>    > ![True vs predicted age](images/true_vs_predicted_plot.png "True vs predicted age")
+>    > The plot in figure 6 shows the true and predicted ages. It can be seen that the points are close.
+>
+{: .hands_on}
+
+> ### {% icon question %} Questions
+>
+> 1. Inspect the plots. What can you say about the predictions?
```
Comment (bebatut) [inline topics/statistics/tutorials/age-prediction_with_ml/tutorial.md]: The question may be before the plot description

File: topics/statistics/tutorials/age-prediction_with_ml/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+> > 1. Figures 4 and 6 show that the prediction is good because the predicted age lie close to the true age.
+> >
+> {: .solution}
+>
+{: .question}
+
+## Conclusion
+
+We can see in figure 4 that we have achieved an r2 score of `0.94` and root mean square score of `3.76` for the test set using gradient boosting regressor. In the study [Jana Naue et al. 2017](https://www.sciencedirect.com/science/article/pii/S1872497317301643?via%3Dihub) as well, a similar root mean square score (`3.93`) is mentioned using random forest regressor. The root mean square score shows the difference in the true and predicted age of humans. The r2 score (`0.94`) is close to the best achievable score of `1.0` which shows that the trained model is good. Overall, the second part of the analysis also shows that using the machine learning tools in Galaxy, we can achieve state-of-the-art predictions mentioned in the recent scientific studies.
+>
```
Comment (bebatut) [inline topics/statistics/tutorials/age-prediction_with_ml/tutorial.md]:
```suggestion
```

File: topics/statistics/tutorials/age-prediction_with_ml/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+> >
+> {: .solution}
+>
+{: .question}
+
+## Conclusion
+
+We can see in figure 4 that we have achieved an r2 score of `0.94` and root mean square score of `3.76` for the test set using gradient boosting regressor. In the study [Jana Naue et al. 2017](https://www.sciencedirect.com/science/article/pii/S1872497317301643?via%3Dihub) as well, a similar root mean square score (`3.93`) is mentioned using random forest regressor. The root mean square score shows the difference in the true and predicted age of humans. The r2 score (`0.94`) is close to the best achievable score of `1.0` which shows that the trained model is good. Overall, the second part of the analysis also shows that using the machine learning tools in Galaxy, we can achieve state-of-the-art predictions mentioned in the recent scientific studies.
+> 
+{:.no_toc}
```
Comment (bebatut) [inline topics/statistics/tutorials/age-prediction_with_ml/tutorial.md]: Do you have a general conclusion about the full tutorial?

**Comment by anuprulez** [comment]
I have fixed the comments. Thanks a lot for reviewing :)
ping @bebatut

**Comment by bebatut** [comment]
@anuprulez I did some fine tuning on formatting. Let me know if it is ok for you.
If yes, I think it is ready to be merged :smile:

**Comment by bgruening** [comment]
thanks @bebatut!
@anuprulez is on holidays the next weeks. I will merge this. Thanks!

**Pull request merged**